### PR TITLE
Add "API Gateway X-Ray Disabled" query for CloudFormation #2365

### LINKED ISSUE #2365: Add "API Gateway X-Ray Disabled" query for CloudFormation
### Platform
CloudFormation

### Provider
AWS

### Description
Xray Tracing is not enabled

### PR DESCRIPTION
**Pull request by pedro-mimoso**
Closes #2365 

**Proposed Changes**

Add "API Gateway X-Ray Disabled" query for CloudFormation

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/cloudFormation/api_gateway_xray_disabled/query.rego [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	resource.Type == "AWS::ApiGateway::Stage"
+	properties := resource.Properties
+	isResFalse(properties.TracingEnabled)
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("Resources.%s.Properties.TracingEnabled", [name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("Resources.%s.Properties.TracingEnabled is false", [name]),
+		"keyActualValue": sprintf("Resources.%s.Properties.TracingEnabled is true", [name]),
```
Comment (Ruben-Silva) [inline assets/queries/cloudFormation/api_gateway_xray_disabled/query.rego:15]: Swapped keyExpected and keyActual? Should be:

"keyExpectedValue": sprintf("Resources.%s.Properties.TracingEnabled is true", [name]),
		"keyActualValue": sprintf("Resources.%s.Properties.TracingEnabled is false", [name]),

File: assets/queries/cloudFormation/api_gateway_xray_disabled/test/negative2.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
+  "AWSTemplateFormatVersion": "2010-09-09",
+  "Resources": {
+    "ProdNeg": {
```
Comment (Ruben-Silva) [inline assets/queries/cloudFormation/api_gateway_xray_disabled/test/negative2.json:4]: Change "ProdNeg" to "ProdNeg2"

File: assets/queries/cloudFormation/api_gateway_xray_disabled/test/positive3.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
+  "AWSTemplateFormatVersion": "2010-09-09",
+  "Resources": {
+    "ProdPos1": {
```
Comment (Ruben-Silva) [inline assets/queries/cloudFormation/api_gateway_xray_disabled/test/positive3.json:4]: Change "ProdPos1" to "ProdPos3"

File: assets/queries/cloudFormation/api_gateway_xray_disabled/test/positive4.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{
+  "AWSTemplateFormatVersion": "2010-09-09",
+  "Resources": {
+    "ProdPos2": {
```
Comment (Ruben-Silva) [inline assets/queries/cloudFormation/api_gateway_xray_disabled/test/positive4.json:4]: Change "ProdPos2" to "ProdPos4"

**Review by Ruben-Silva — CHANGES_REQUESTED** [review]

**Review by Ruben-Silva — DISMISSED** [review]
Looks good to me!

**Review by ruigomescx — APPROVED** [review]

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**